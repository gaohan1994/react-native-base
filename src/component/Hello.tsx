import React from 'react';
import {
    Button,
    StyleSheet,
    Text,
    View,
} from 'react-native';
import HelloClass from '../class/hello';

interface Props {}
interface State {
    value: number;
}

class Hello extends React.Component <Props, State> {

    constructor (props: Props) {
        super(props);
        this.state = {
            value: 0
        };
    }

    public testClassHandle = () => {
        HelloClass.baseMethod();
    }

    public onDecrementHandle = (): void => {
        this.setState({
            value: this.state.value - 1
        });
    }

    public onIncrementHandle = (): void => {
        this.setState({
            value: this.state.value + 1
        });
    }

    render (): React.ReactNode {

        const { value } = this.state;

        return (
            <View>
                <Text>Hello, Gaohan, {value} times</Text>

                <View>
                    <Button
                        title="-"
                        onPress={this.onDecrementHandle}
                        color="red"
                        accessibilityLabel="decrement"
                    />
                </View>

                <View>
                    <Button
                        title="+"
                        onPress={this.onIncrementHandle}
                        color="blue"
                        accessibilityLabel="increment"
                    />
                </View>
            </View>
        );
    }
}

export default Hello;