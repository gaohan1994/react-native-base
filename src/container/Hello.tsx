/**
 * created by Ghan 2018.7.31
 * 首页
 */

import React from 'react';
import {
    Button,
    StyleSheet,
    Text,
    View,
    StatusBar,
    // Platform
} from 'react-native';
import HelloClass from '../class/hello';

interface Props {}

interface State {
    value: number;
}

class Hello extends React.Component <Props, State> {

    constructor (props: Props) {
        super(props);
        this.state = {
            value: 0
        };
    }

    public testClassHandle = () => {
        HelloClass.baseMethod();
    }

    public onDecrementHandle = (): void => {
        this.setState({
            value: this.state.value - 1
        });
    }

    public onIncrementHandle = (): void => {
        this.setState({
            value: this.state.value + 1
        });
    }

    render (): React.ReactNode {

        const { value } = this.state;
        console.log('__DEV__', __DEV__);
        return (
            <View style={styles.container}>
                <Text>Hello, Gaohan, {value} times</Text>
                <Text>
                    {__DEV__ === true
                    ? 'dev'
                    : 'pro'}
                </Text>
                <View>
                    <Button
                        title="-"
                        onPress={this.onDecrementHandle}
                        color="red"
                        accessibilityLabel="decrement"
                    />
                </View>

                <View>
                    <Button
                        title="+"
                        onPress={this.onIncrementHandle}
                        color="blue"
                        accessibilityLabel="increment"
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#F8F8F8',
        position: 'relative'
    },
});

export default Hello;