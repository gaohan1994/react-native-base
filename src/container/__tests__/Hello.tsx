import React from 'react';
import renderer from 'react-test-renderer';
import Hello from '../Hello';

describe('Hello test begin', () => {

  it('should render Hello', () => {
      const tree = renderer.create(<Hello/>).toJSON();
      expect(tree).toMatchSnapshot();
  });
});
