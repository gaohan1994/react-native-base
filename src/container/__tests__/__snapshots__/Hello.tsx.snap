// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello test begin should render Hello 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello, Gaohan, 
    0
     times
  </Text>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="decrement"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="increment"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "blue",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
